pina/model/layers/convolution_2d.py

```diff
@@ -0,0 +1,292 @@
+            # shape: [sum(numb_points_insiede), filter_dim + 1]
+            point_stride = x[indeces]
+
+            # mapping points in filter domain
+            map_points_(point_stride[..., :-1], current_stride)
+
+            # extracting points for each channel
+            point_stride_channel = point_stride.split(numb_points_insiede)
+
+            # appending in list for later use
+            mapped_points.append(point_stride_channel)
+            indeces_channels.append(numb_points_insiede)
+
+        # stacking input for passing to neural net
+        mapping = map(torch.cat, zip(*mapped_points))
+        stacked_input = tuple(mapping)
+        indeces_channels = tuple(zip(*indeces_channels))
+
+        return stacked_input, indeces_channels
+
+    def _find_index(self, X):
+        """Private method to extract indeces for convolution
+
+        :param X: input tensor, as in ContinuousConv2D docstring
+        :type X: torch.tensor
+        """
+        index = []
+        for _, current_stride in enumerate(self._stride):
+
+            tmp = check_point(X, current_stride, self._dim)
+            index.append(tmp)
+
+        self._index = index
+
+    def forward(self, X):
+        """Forward pass in the FFConv layer
+
+        :param x: input data ( input_numb_field x N x filter_dim )
+        :type x: torch.tensor
+        :return: feed forward convolution ( output_numb_field x N x filter_dim )
+        :rtype: torch.tensor
+        """
+        # batch dimension
+        batch_dim = X.shape[0]
+        conv = self._make_grid(batch_dim)
+
+        # extracting indeces for convolution on batches
+        # TODO: if we want to optimize and we know that
+        # the grid is the same for all samples in the
+        # training set, we can do something like call this
+        # function just once when we are in .train() mode
+        self._find_index(X)
+
+        for batch_idx, x in enumerate(X):
+
+            # extract mapped points
+            stacked_input, indeces_channels = self._extract_mapped_points(
+                batch_idx, self._index, x)
+
+            # for each output numb field
+            tot_dim = self._output_numb_field * self._input_numb_field
+            res_tmp = []
+            for idx_conv in range(tot_dim):
+                # compute convolution
+                idx = idx_conv % self._input_numb_field
+                # extract input for each channel
+                single_channel_input = stacked_input[idx]
+                # extract filter
+                net = self._net[idx_conv]
+                # caculate filter value
+                staked_output = net(single_channel_input[..., :-1])
+                # perform integral for all strides in one channel
+                integral = self._integral(staked_output,
+                                          single_channel_input[..., -1],
+                                          indeces_channels[idx])
+                res_tmp.append(integral)
```

Comment:
this part is quite difficult to read, I personally lost the meaning of such lines. Also the comments are not really consistent with instructions imho

Reply:
solved in [a531e0e](https://github.com/mathLab/PINA/pull/69/commits/a531e0e9a2e135bae61dca44e4c6a6c18a76fb6b)